module/documents/actor/trait.mjs

```diff
@@ -269,8 +268,19 @@ export function getBaseItem(identifier, { indexOnly=false, fullItem=false }={})
   if ( !packObject ) return;
 
   // Build the extended index and return a promise for the data
-  const promise = packObject.getIndex({ fields: traitIndexFields() }).then(index => {
+  const fields = traitIndexFields();
+  const promise = packObject.getIndex({ fields }).then(index => {
     const store = index.reduce((obj, entry) => {
+      for (const field in fields) {
+        const val = foundry.utils.getProperty(entry, field);
+        if (field !== "system.type.value" && val !== undefined) {
+          foundry.utils.setProperty(entry, "system.type.value", val);
+          foundry.utils.logCompatibilityWarning(
+            `The '${field}' property has been deprecated in favor of a standardized \`system.type.value\` property.`,
+            { since: "DnD5e 2.5.0", until: "DnD5e 2.6" }
```

Comment:
We usually give at least 2 versions for changes. I've also added the `once` option that was introduced in v12 to reduce these warnings in the future.

```suggestion
            { since: "DnD5e 2.5", until: "DnD5e 2.7", once: true }
```